base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -673,6 +694,143 @@ where
+                fee_per_gram,
+                None,
+                message.clone(),
+                script!(PushPubKey(Box::new(dest_pubkey.clone()))),
+            )
+            .await?;
+        let tx_id = stp.get_tx_id()?;
+        if !stp.is_single_round_message_ready() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round message not ready".to_string(),
+            ));
+        }
+
+        let _ = stp
+            .build_single_round_message()
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        if !stp.is_collecting_single_signature() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round signature not being collected".to_string(),
+            ));
+        }
+        self.output_manager_service
+            .confirm_pending_transaction(tx_id)
+            .await
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+
+        // Prepare receiver part of the transaction
+
+        // Diffie-Hellman shared secret `k_Ob * K_Sb = K_Ob * k_Sb` results in a public key, which is converted to
+        // bytes to enable conversion into a private key to be used as the spending key
+        // TODO: Add a standardized Diffie-Hellman method to the tari_crypto library, then come back and use it here
+        let script_offset_private_key = stp
+            .get_recipient_script_offset_private_key(0)
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        let spending_key = PrivateKey::from_bytes((script_offset_private_key.clone() * dest_pubkey.clone()).as_bytes())
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+
+        let sender_message = TransactionSenderMessage::new_single_round_message(stp.get_single_round_message()?);
+        let rtp = ReceiverTransactionProtocol::new(
+            sender_message,
+            PrivateKey::random(&mut OsRng),
+            spending_key,
+            OutputFeatures::default(),
+            &self.resources.factories,
+        );
+
+        let recipient_reply = rtp.get_signed_data()?.clone();
+
+        // Start finalizing
+
+        stp.add_single_recipient_info(recipient_reply, &self.resources.factories.range_proof)
+            .map_err(|e| TransactionServiceProtocolError::new(tx_id, TransactionServiceError::from(e)))?;
+        if !stp.is_finalizing() {
+            return Err(TransactionServiceError::OneSidedTransactionError(
+                "Sender protocol single round tx not being finalized".to_string(),
+            ));
+        }
+
+        // Finalize
+
+        stp.finalize(KernelFeatures::empty(), &self.resources.factories)
+            .map_err(|e| {
+                error!(
+                    target: LOG_TARGET,
```

Comment:
Is there a reason that for the extra log here?

Reply:
Yes, it is to help with system-level debugging.

Reply:
but won't it get logged by a caller of this method? Adding in extra error logging clouds the important code

Reply:
In my experience add the log where it is needed